src/scry/symbol.cr

```diff
@@ -115,4 +116,26 @@ module Scry
       ResponseMessage.new(@text_document.id, visitor.symbols)
     end
   end
+
+  class WorkspaceSymbolProcessor
+    def initialize(@msg_id : Int32 | String, @root_path : String, @query : String)
+      @all_files = Dir.glob("#{root_path}/**/*.cr")
```

Comment:
Why doesn't this use the workspace, and find files based on prelude and requires?

Reply:
> and find files based on prelude and requires?

What do you mean? :sweat_smile: This will get all "workspace" files to analyze all "workspace" symbols

Reply:
Oh ok, so this symbol feature is just to search in your project, not everywhere (stdlib, other shards, etc) ?

Reply:
Ok it's fine then. I think I'd even restrict to `src`, to not show implementation detail symbols from shards

Reply:
@bew Well, I guess I like your idea of listing stdlib symbols, so, this feature will support listing symbols from:

1. Your code
2. Your shards
3. Crystal stdlib

![screenshot_20180505_181554](https://user-images.githubusercontent.com/3067335/39668324-5ad45952-5090-11e8-89b5-dd16bedff017.png)

And works very fast!!! :tada: and stdlib is memoized :sparkles: